src/main/scala/scala/tools/refactoring/common/TreeTraverser.scala

```diff
@@ -119,7 +119,15 @@ trait TreeTraverser extends TracingImpl {
       }
 
       val treesFromType = {
-        val tpeWithoutPrefix = tpe.trimPrefix(tpe.toString).split("\\.") match {
+        def splitUntilTypeParameterBracket(acc: Array[String])(tpe: String): Array[String] = {
+          if (tpe.isEmpty)
+            acc
+          else if (!tpe.contains(".") || tpe.contains(".") && tpe.contains("[") && tpe.indexOf("[") < tpe.indexOf("."))
+            acc :+ tpe
```

Comment:
This one is unfair :), regexp for catching imports doesn't expect ` in
package name. But ok, will be added.

W

2016-04-17 8:04 GMT+02:00 Matthias Langer <notifications@github.com>:

> In src/main/scala/scala/tools/refactoring/common/TreeTraverser.scala
> <https://github.com/scala-ide/scala-refactoring/pull/158#discussion_r59978819>
> :
>
> > @@ -119,7 +119,15 @@ trait TreeTraverser extends TracingImpl {
> >        }
> >
> >        val treesFromType = {
> > -        val tpeWithoutPrefix = tpe.trimPrefix(tpe.toString).split("\\.") match {
> > +        def splitUntilTypeParameterBracket(acc: Array[String])(tpe: String): Array[String] = {
> > +          if (tpe.isEmpty)
> > +            acc
> > +          else if (!tpe.contains(".") || tpe.contains(".") && tpe.contains("[") && tpe.indexOf("[") < tpe.indexOf("."))
> > +            acc :+ tpe
>
> I'm sorry, but you provoked me with your comment about Sauron :-) Try
> organizing imports on
>
> object `Eye[` {
>   object `.Of` {
>     object Sauron
>     object TheBeholder
>   }
> }
> class Bug {
>   import `Eye[`.`.Of`.TheBeholder
>   import `Eye[`.`.Of`.Sauron
>
>   val x = Sauron
> }
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/scala-ide/scala-refactoring/pull/158/files/ff405061b5485d941a88697c3c21743b4677b601#r59978819>
>